### PR TITLE
feature(bigquery): add quickstart

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6116)

### CODE REVIEW

**Comment by devjgm** [comment]
I think this PR needs to be rebased. Sorry about that.

File: google/cloud/bigquery/quickstart/quickstart.cc [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+}  // namespace
+
+int main(int argc, char* argv[]) try {
+  if (argc != 3) {
+    std::cerr << "Usage: " << argv[0] << " <project-id> <table>\n";
+    return 1;
+  }
+
+  // project_id should be in the format "projects/<your-gcp-project>"
```
Comment (coryan) [inline google/cloud/bigquery/quickstart/quickstart.cc:34]: For other resources (Bigtable tables, Spanner databases, etc.) we call the fully qualified path `$resource name` (e.g. table name, database name), and the unqualified id, `$resource id` (e.g. table id, database id).

In other words, consider `std::string const project_name = argv[1];` and change the comments accordingly.

**Review by coryan — COMMENTED** [review]
Do you want this to be compiled and tested in one of the integration tests?  Right now it is not, the quickstart guides usually have their own build steps.  Chat with @devjgm about whether you should include this *only* in the GCB builds or if we want to add them to the other builds too.

Note that there are separate `quickstart` scripts for  CMake and Bazel on each of Windows, macOS and Linux.

**Comment by devjgm** [comment]
> Do you want this to be compiled and tested in one of the integration tests? Right now it is not, the quickstart guides usually have their own build steps. Chat with @devjgm about whether you should include this _only_ in the GCB builds or if we want to add them to the other builds too.
> 
> Note that there are separate `quickstart` scripts for CMake and Bazel on each of Windows, macOS and Linux.

Happy to chat. FTR, for GCB-only builds, integration tests can go in https://github.com/googleapis/google-cloud-cpp/blob/master/ci/cloudbuild/builds/integration.sh

I don't have quickstarts running yet, but they are just about next on my list, and I expect to have those running in soon. However, I wouldn't want to block @scotthart. If Scott thinks that doing something w/ GCB would be _helpful_ for him, that's great. Otherwise, if doing things all in kokoro is faster for him, that's fine too.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6116)* status: 0 of 7 files reviewed, 2 unresolved discussions (waiting on @coryan)

___
*[google/cloud/bigquery/quickstart/quickstart.cc, line 34 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6116#-MWf2QGbKoPKDiRe_6uD-r1-34:-MWjDp0n6MSLs9E7x6mR:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/9624b73109cfbc781690994db74168c64c78e5cf/google/cloud/bigquery/quickstart/quickstart.cc#L34)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

For other resources (Bigtable tables, Spanner databases, etc.) we call the fully qualified path `$resource name` (e.g. table name, database name), and the unqualified id, `$resource id` (e.g. table id, database id).

In other words, consider `std::string const project_name = argv[1];` and change the comments accordingly.

</blockquote></details>

Done.
___
*[google/cloud/bigquery/quickstart/quickstart.cc, line 53 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6116#-MWf2QGbKoPKDiRe_6uD-r1-53:-MWjDqcKETktehlnGPkV:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/9624b73109cfbc781690994db74168c64c78e5cf/google/cloud/bigquery/quickstart/quickstart.cc#L53)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

The project is hard-coded, you probably want to use `argv[1]`.
</blockquote></details>

Done.

**Review by coryan — COMMENTED** [review]
I think you need to ignore this directory here:

https://github.com/googleapis/google-cloud-cpp/blob/1ee16f3c9a4f88c32e0acc1360ae7365dab420d1/.bazelignore#L9

> Reviewed 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6116#-:-MWjE8AkDs-YU_5XDeSy:b-v36c3i)* status: 2 of 7 files reviewed, all discussions resolved (waiting on @scotthart)

**Review by coryan — APPROVED** [review]
> Reviewed 5 of 7 files at r1, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6116)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @scotthart)
